packages/core-components/src/layout/Sidebar/Items.tsx

```diff
@@ -149,29 +155,123 @@ const useStyles = makeStyles<BackstageTheme>(
+      },
+      arrows: {
         position: 'absolute',
-        right: 0,
+        right: 10,
       },
       selected: {
         '&$root': {
           borderLeft: `solid ${selectedIndicatorWidth}px ${theme.palette.navigation.indicator}`,
           color: theme.palette.navigation.selectedColor,
         },
-        '&$closed': {
+        buttonItem: {
+          background: 'none',
+          border: 'none',
+          width: 'auto',
+          margin: 0,
+          padding: 0,
+          textAlign: 'inherit',
+          font: 'inherit',
+        },
+        closed: {
           width: drawerWidthClosed,
+          justifyContent: 'center',
+        },
+        open: {
+          width: drawerWidthOpen,
+        },
+        label: {
+          // XXX (@koroeskohr): I can't seem to achieve the desired font-weight from the designs
+          fontWeight: 'bold',
+          whiteSpace: 'nowrap',
+          lineHeight: 'auto',
+          flex: '3 1 auto',
+          width: '110px',
+          overflow: 'hidden',
+          'text-overflow': 'ellipsis',
         },
-        '& $closedItemIcon': {
-          paddingRight: selectedIndicatorWidth,
+        iconContainer: {
+          boxSizing: 'border-box',
+          height: '100%',
+          width: iconContainerWidth,
+          marginRight: -theme.spacing(2),
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
         },
-        '& $iconContainer': {
-          marginLeft: -selectedIndicatorWidth,
+        searchRoot: {
+          marginBottom: 12,
+        },
+        searchField: {
+          color: '#b5b5b5',
+          fontWeight: 'bold',
+          fontSize: theme.typography.fontSize,
+        },
+        searchFieldHTMLInput: {
+          padding: `${theme.spacing(2)} 0 ${theme.spacing(2)}`,
+        },
+        searchContainer: {
+          width: drawerWidthOpen - iconContainerWidth,
+        },
+        secondaryAction: {
+          width: theme.spacing(6),
+          textAlign: 'center',
+          marginRight: theme.spacing(1),
         },
       },
     };
   },
   { name: 'BackstageSidebarItem' },
 );
 
+const useLocationMatch = (
+  submenu: React.ReactElement<SidebarSubmenuProps>,
+  locationPathname: string,
+): boolean =>
+  // Evaluates the routes of the SubmenuItems & nested DropdownItems.
+  // The reeveluation is only triggered, if the `locationPathname` changes, as `useElementFilter` uses memorization
+  useElementFilter(
+    submenu.props.children,
+    elements => {
+      let active = false;
+      elements
+        .getElements()
+        .forEach(
+          ({
+            props: { to, dropdownItems },
+          }: {
+            props: Partial<SidebarSubmenuItemProps>;
+          }) => {
+            if (!active) {
+              if (dropdownItems?.length) {
+                dropdownItems.forEach(
+                  ({ to: _to }) =>
+                    (active = active || locationPathname.includes(_to)),
```

Comment:
hmmm, `includes` feels like an oddball here (and below) doesn't it - at the very least `startsWith` could perhaps be used, but makes me think that some kind of react router matcher thingy might exist to help with this

Reply:
saw that there was another location matching implemented (#8378) - used that now instead - not sure it is optimal, but we have one source of truth to iterate on.